monitor/flowgen/bridging/bridging.go

```diff
@@ -37,6 +39,19 @@ func newJob(networkID netconf.ID, backends ethbackend.Backends, conf flowConfig,
 		return types.Job{}, errors.Wrap(err, "src chain backend")
 	}
 
+	addrs, err := contracts.GetAddresses(ctx, networkID)
+	if err != nil {
+		return types.Job{}, errors.New("contract addresses")
+	}
+
+	solverAddr := eoa.MustAddress(networkID, eoa.RoleSolver)
+	checker := solver.NewChecker(
```

Comment:
not a fan of simply extracting the logic from solver into monitor. Encapsulation of logic and ownership is important. Why not call the actual API? Then we test test infra at the same time.